app/src/main/java/com/github/se/orator/ui/offline/OfflineRecordingScreen.kt

```diff
@@ -109,30 +109,62 @@ fun OfflineRecordingScreen(
             horizontalAlignment = Alignment.CenterHorizontally) {
               Box(
                   contentAlignment = Alignment.Center,
-                  modifier =
-                      Modifier.size(AppDimensions.logoSize)
-                          .testTag("MicIconContainer")) { // // should be 203.dp
-                    Image(
-                        painter = painterResource(id = R.drawable.bckgrnd_blobs),
-                        contentDescription = "Background",
-                        modifier = Modifier.size(AppDimensions.logoSize).testTag("BackgroundBlob"))
-                    Icon(
-                        imageVector = Icons.Filled.Mic,
-                        contentDescription = "Microphone",
-                        modifier =
-                            Modifier.size(AppDimensions.iconSizeMic)
-                                .scale(if (isRecording) scale else 1f)
-                                .clickable {
-                                  if (permissionGranted) {
-                                    speakingViewModel.onMicButtonClicked(true)
-                                  } else {
-                                    permissionLauncher.launch(Manifest.permission.RECORD_AUDIO)
-                                  }
+                  modifier = Modifier.size(AppDimensions.logoSize).testTag("MicIconContainer")) {
+                    MicrophoneButton(
+                        viewModel,
+                        collState,
+                        permissionGranted,
+                        context,
+                        funRec = {
+                          // what to do when user begins to record a file
+                          if (analysisState.value == SpeakingRepository.AnalysisState.IDLE ||
+                              analysisState.value == SpeakingRepository.AnalysisState.IDLE) {
+                            File(context.cacheDir, "${viewModel.interviewPromptNb.value}.mp3")
+                                .also {
+                                  recorder.startRecording(it)
+                                  audioFile = it
+                                }
+                            Log.d(
+                                "aa",
+                                "now transcribing to: ${viewModel.interviewPromptNb.value}.mp3")
+                            analysisState.value = SpeakingRepository.AnalysisState.RECORDING
+                          }
+                          // what to do when user finishes recording a file
+                          else if (analysisState.value ==
```

Comment:
This section could be refactored using a when expression instead of if-else statements. It would improve readability, make the code more concise, and provide a structured way to handle additional states in the future.

Reply:
I added an else statement with a log in case we are in the FINISHED state to debug.